shard/src/ext/latex_cog.py

```diff
@@ -63,19 +63,33 @@ async def latex(self, ctx, *latex):
             tex = await create_subprocess_exec('latex', '-halt-on-error', '-no-shell-escape',
                                                '-interaction batchmode', 'out.tex',
                                                cwd=work_dir,
-                                               stdout=DEVNULL,
+                                               stdout=PIPE,
                                                stderr=DEVNULL,
                                                close_fds=True)
 
             try:
-                await wait_for(tex.wait(), timeout=3)
+                stdout, _ = await wait_for(tex.communicate(), timeout=3)
             except TimeoutError:
                 await ctx.send("Compilation took too long")
                 tex.kill()
                 return
 
             if not os.path.isfile(os.path.join(work_dir, 'out.dvi')):
-                await ctx.send("Compilation failed")
+                output = stdout.decode('ascii')
+
+                # The only latex output that we want comes after the first '!'
+                # Pdflatex includes a bunch of boilerplate output at the beginning
+                # about the state of the environment that we don't really need.
+                # In addition, the last two lines of output are just saying where
+                # pdflatex stored the logs to but the discord user doesn't have
+                # access to those so just get rid of them.
+                error_msg = output[output.index('!'):]
+                error_msg = "\n".join(error_msg.split('\n')[:-3])
+
+                embed = Embed(title="Latex Compilation Error", description="Something went wrong")
+                embed.add_field(name="Latex Code", value=latex, inline=False)
+                embed.add_field(name="Compiler Error", value=error_msg, inline=False)
+                await ctx.send(embed=embed)
```

Comment:
does this always output something?  if there's any possibility that this won't send (cause of a malfromed embed or w/e) we should have a fallback message.

Reply:
It should always have some output. If the code goes through that branch, it means that the compiler tried to compile the latex and failed within the time limit. The only reason that should happen is due to a syntax error which would mean there is compiler output that can be sent back to the user.